### PR TITLE
chore: fix storybook styles, imports, typescript config and dependencies (#5319

### PR DESCRIPTION
**Pull request by ovflowd**
This PR does a few fixes related to TypeScript config, missing dependencies (and their types), the old styles being loaded on Storybook and others.

- Move all old styles to `styles/old`
- Add missing dependencies
- Optimised TypeScript config file
- Added (partial) base styles from `nodejs.dev`
- Fixed Storybook import of Styles
- Changed `sr-only` components to `aria-label`
- Added OpenSans font-family
- Updated `global.d.ts` with global JSON module type
- Fixes Banner Component by removing old styles and updated the Stories

### CODE REVIEW

**Comment by ovflowd** [comment]
Hey, y'all (@nodejs/website) can we fast-track this PR as it does some fixes that we need on some PRs? Thank you!

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -26,7 +30,22 @@ const preview: Preview = {
...
   Story => (
     <ThemeProvider>
-      <App Component={Story} pageProps={pageProps} />
+      <SiteProvider>
+        <LocaleProvider i18nData={pageProps.i18nData}>
+          <NodeDataProvider nodeVersionData={pageProps.nodeVersionData}>
+            <style>
+              {`
+                body {
+                  font: 400 20px/1.5 ${openSans.style.fontFamily},
```
Comment (AugustinMauroy) [inline .storybook/preview.tsx]: For font size we should use this var
```
--font-size-body1: 1.6rem;
```

File: pages/_app.mdx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { NodeDataProvider } from '../providers/nodeDataProvider';
 import { LocaleProvider } from '../providers/localeProvider';
 import { SiteProvider } from '../providers/siteProvider';
 import sourceSansPro from '../util/sourceSansPro';
+
 import '../styles/styles.scss';
 import '../styles/tokens.scss';
```
Comment (AugustinMauroy) [inline pages/_app.mdx]: We can remove this it's doesn't use.
```suggestion
```

File: components/Common/Banner/index.module.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@import 'styles/_variables';
+@import 'styles/old/_variables';
```
Comment (AugustinMauroy) [inline components/Common/Banner/index.module.scss]: We could update this component to not use the old style. And anyway we will have to do it at some point

Comment (ovflowd) [reply components/Common/Banner/index.module.scss]: Not the scope of this PR.

Comment (AugustinMauroy) [reply components/Common/Banner/index.module.scss]: sure, but should we update that ?

Comment (ovflowd) [reply components/Common/Banner/index.module.scss]: Okay, I've added in this PR the changes

Comment (AugustinMauroy) [reply components/Common/Banner/index.module.scss]: Oh ok 😁

**Review by AugustinMauroy — APPROVED** [review]
LGTM !
Thanks Claudio for this fix.

**Comment by ovflowd** [comment]
I'm merging this as there are a few issues right now on the `major/website-redesign` branch.

I take full responsibility. Please let me know if anyone here belives this should not have landed! 🙇

**Pull request merged**